### PR TITLE
fix potential nil pointer error

### PR DESCRIPTION
**Pull request by yiminc**
fix potential nil pointer bug

### CODE REVIEW

File: internal/internal_activity.go [hunk 1]
```diff
@@ -175,7 +175,7 @@ func getValidatedLocalActivityOptions(ctx Context) (*executeLocalActivityParams,
 
 func validateFunctionArgs(f interface{}, args []interface{}, isWorkflow bool) error {
 	fType := reflect.TypeOf(f)
-	if fType.Kind() != reflect.Func {
+	if fType == nil || fType.Kind() != reflect.Func {
```
Comment (madhuravi) [inline internal/internal_activity.go]: Curious when will this be nil? If f is nil? In that case, should we return an error? Is nil not a valid argument?

File: internal/internal_activity.go [hunk 2]
```diff
@@ -240,9 +240,17 @@ func getValidatedActivityFunction(f interface{}, args []interface{}) (*ActivityT
 	return &ActivityType{Name: fnName}, input, nil
 }
 
+func getReflectTypeKind(fType reflect.Type) reflect.Kind {
```
Comment (madhuravi) [inline internal/internal_activity.go]: you can just name this "kind", might be simpler to read. We can infer it is for reflect.Type from the function argument.

File: internal/internal_activity.go (hunk 1 above)
Comment (yiminc) [reply internal/internal_activity.go]: If you pass nil as activity in ExecuteLocalActivity or ExecuteLocalActivity.
I run into this accidentally in a unit test. I defined an dummy activity in the test that takes no parameter and returns nil. I should have call workflow.ExecuteLocalActivity(ctx, dummyActivity), but I used dummyActivity() which returns nil and did workflow.ExecuteLocalActivity(ctx, dummyActivity()). The workflow panic with nil pointer error but have no clue where the error comes from.

File: internal/internal_activity.go (hunk 2 above)
Comment (yiminc) [reply internal/internal_activity.go]: sure

File: internal/internal_activity.go [hunk 3]
```diff
@@ -240,9 +240,17 @@ func getValidatedActivityFunction(f interface{}, args []interface{}) (*ActivityT
 	return &ActivityType{Name: fnName}, input, nil
 }
 
+func getKind(fType reflect.Type) reflect.Kind {
+	fTypeKind := reflect.Invalid
+	if fType != nil {
+		fTypeKind = fType.Kind()
+	}
+	return fTypeKind
+}
```
Comment (mfateev) [inline internal/internal_activity.go]: What about:

```
  if fType == nil {
    return reflect.Invalid
  }
  return fType.Kind()
```

**Pull request merged**